homeassistant/auth/__init__.py

```diff
@@ -256,20 +256,24 @@ def async_create_access_token(self,
         return await auth_provider.async_credential_flow(context)
 
     async def _async_finish_login_flow(
-            self, context: Optional[Dict], result: Dict[str, Any]) \
-            -> Optional[Union[models.User, models.Credentials]]:
+            self, flow: data_entry_flow.FlowHandler, result: Dict[str, Any]) \
+            -> Dict[str, Any]:
         """Return a user as result of login flow."""
         if result['type'] != data_entry_flow.RESULT_TYPE_CREATE_ENTRY:
-            return None
+            result.pop('result', None)
```

Comment:
There can't be a `result` entry in the ABORT or SHOW_FORM results?

Reply:
Not in current design (before this PR), we only set `result[result]` when `result['type'] == RESULT_TYPE_CREATE_ENTRY`

Reply:
sorry, that's what I meant 😅 

There can't be a `result` key in the `result` dict because it's not from the CREATE_ENTRY type.

Reply:
I see, you may right, this statement will always fell into do nothing.

Probably we don't need this check at all, we can make sure the call back only be called when `type==CREATE_ENTRY`

Reply:
No, we need to call it as it's up to the implementation of the Flow manager to decide how to handle it.